src/sentry/migrations/0748_increase_project_slug_max_length.py

```diff
@@ -0,0 +1,34 @@
+# Generated by Django 5.0.7 on 2024-08-07 17:15
+
+from django.db import migrations
+
+import sentry.db.models.fields.slug
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
```

Comment:
should we make this a post deployment migration? not quite sure how many projects there are or the scale of this migration.

Reply:
No, post deployment is for backfills and large indexes only, not thing that modify the schema.

Increasing the side of this column should be fast. If it's not, we have a statement timeout that will cause the migration to break within a few seconds.

I'd say that just make sure you're around to monitor this and revert it if it won't run - it should be ok though.

Reply:
We might get hit with an index rebuild which could throw off query performance for slug lookups. The [postgres docs](https://www.postgresql.org/docs/current/sql-altertable.html#SQL-ALTERTABLE-DESC-SET-DATA-TYPE) state

> When this form is used, the column's statistics are removed, so running [ANALYZE](https://www.postgresql.org/docs/current/sql-analyze.html) on the table afterwards is recommended.

We have had [an incident](https://www.notion.so/sentry/INC-654-Database-migration-on-ApiTokenReplica-resulted-in-down-Sentry-7188774d88a848ffb05d7e092e0234f5) where a column resize invalided stats and caused an outage. We could have a similar outcome here.

Reply:
How do you run ANALYZE, I can't figure it out from the [example in docs](https://www.notion.so/sentry/PostgreSQL-d2be3f78723d4b6fad12001f3522a5d2?pvs=4#701c3bab6cc74ee58309f773d361de1f)

Reply:
Good call @markstory, I'd forgotten about that. Maybe in that case, we should make this a post_deploy migration and trigger the analyze as part of it, or even have ops run the entire thing manually for us.

I doubt we can run the ANALYZE as part of the deploy, it'll very likely time out.

Reply:
I switched it to post deploy, do I have to include the ANALYZE as part of the migration code?